static/src/javascripts/projects/common/modules/commercial/build-page-targeting.js

```diff
@@ -159,6 +158,19 @@ const getUrlKeywords = (pageId) => {
     return [];
 };
 
+const flattenDeep = arr => {
+    let flattened = [];
+    for (const a of arr) {
+        if (Array.isArray(a)) {
+            flattened = [...flattened, ...a];
+            flattened = [...flattenDeep(flattened)]
+        } else {
+            flattened.push(a);
+        }
+    }
+    return flattened;
+}
```

Comment:
Optional:
Might be a silly question but can we avoid deep flattening?
i.e. Can we transform the the input data early to a format where it does not need flattening and we just concat?

Reply:
Not silly at all. I've tried to but the same data are used in other places and they were expecting that format. 
I did change the logic to concat though and [it performs a bit better ](https://jsbench.me/8gkmxhh8wf/2)